src/apps/datasets/models.py

```diff
@@ -60,12 +63,70 @@ def save(self, *args, **kwargs):
             self.name = f"{self.created_by.username} - {self.type}"
         return super().save(*args, **kwargs)
 
+    @property
+    def in_use(self):
+        from tasks.models import Task
+        tasks = Task.objects.filter(Q(ingestion_program=self) | Q(input_data=self) | Q(reference_data=self) | Q(
+            scoring_program=self)).prefetch_related('phases')
+        phases_from_tasks = [phase for task in tasks for phase in task.phases.all()]
+        from competitions.models import Phase
+        phases = Phase.objects.filter(Q(ingestion_program=self) | Q(input_data=self) | Q(reference_data=self) | Q(
+            scoring_program=self)).prefetch_related('competition')
+        print(tasks)
+        print(phases_from_tasks)
+        print(phases)
+        task_competitions = [phase.competition.pk for phase in phases_from_tasks if phase.competition]
+        phase_competitions = [phase.competition.pk for phase in phases if phase.competition]
+        competition_set = list(set(task_competitions + phase_competitions))
+        is_used = bool(competition_set)
+        return {'value': is_used,
+                'competitions': competition_set}
+
     def __str__(self):
         return self.name or ''
 
+    def get_chahub_endpoint(self):
+        return "datasets/"
+
+    def clean_chahub_data(self, temp_data):
+        data = temp_data
+        for key in list(data):
+            if not data[key] or data[key] == '':
+                data.pop(key, None)
```

Comment:
If you add a `continue` here, you don't need to do the `data.get()`s down below? Since you don't actually need to check the other `if`s if you have popped it from the data.